### DISCUSSION TITLE
using preload() and optionsLimit() in Select relationship

### DISCUSSION THREAD

**Discussion by wychoong** [filamentphp/filament#13511] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.92

### How can we help you?

I have a select field with following configuration
```php
Forms\Components\Select::make('customer_id')
          ->visibleOn('create')
          ->relationship('customer', 'name')
          ->getOptionLabelFromRecordUsing(fn ($record) => "....")
          ->allowHtml()
          ->preload()
          ->required()
          ->optionsLimit(50)
          ->searchable(['name', 'mobile', 'email']),
```

In my use case, I wanted to have the Select to show the options on click without waiting server request. A few issues arise which
- `preload`, but since its searchable it never uses the preloaded options
- `optionsLimit` is only applied to `relationship->getSearchResultUsing` https://github.com/filamentphp/filament/blob/821635f73a98401b156ba8c4c76356754f0ed62e/packages/forms/src/Components/Select.php#L717-L726 , should it be applied to `relationship->options` if it's searchable? https://github.com/filamentphp/filament/blob/821635f73a98401b156ba8c4c76356754f0ed62e/packages/forms/src/Components/Select.php#L769-L777 
- when open the dropdown, a request to server is called. however @danharrin mentioned it as intended https://github.com/filamentphp/filament/issues/7251#issuecomment-1653740959 , but would like to be optimized, currently a query is made but the generated options are never used, is it possible on first load it is skipped?

would like to see if this can be improved, v3 or v4

**Comment by danharrin** [comment, marked as answer]
If a function is passed to `options()`, they are loaded when the select is opened.

If you pass a static array, it will use that.

`relationship()` passes a function.

If you pass a static array after `relationship()`, it will override the function.
